src/components/modals/RedeemModal.tsx

```diff
@@ -93,11 +137,10 @@ export default function RedeemModal({
 
         if (onCancel) onCancel()
       }}
-      okText={`Burn ${formattedNum(redeemAmount, {
-        precision: 2,
-      })} ${tokenSymbol ?? 'tokens'} for ETH`}
+      okText={buttonText}
       okButtonProps={{
-        disabled: !redeemAmount || parseInt(redeemAmount) === 0,
+        disabled:
+          !redeemAmount || parseInt(redeemAmount) === 0 || invalidRedeemAmount,
```

Comment:
maybe we shouldn't disable the button for `invalidRedeemAmount`, since there won't be anything telling the user why the button is disabled? we could just rely on your validator for that

Reply:
Yep good call, makes sense to me